### ISSUE TITLE
[clang-format] Wrong spacing around arithmetic operator?

### ISSUE THREAD

**Issue by prj-** [llvm/llvm-project#102727]
I'm not sure these two should by formatted differently?
```console
$ clang-format bug.c
v[num] = ((PetscReal)(numj)*HxdHy + (PetscReal)(numi)*HydHx);
v[num] = ((PetscReal)numj * HxdHy + (PetscReal)numi * HydHx);
```
This is with both version 18 and 20.

**Comment by owenca** [comment]
Why the parentheses surrounding `numi` and `numj`? Is the assignment in a macro definition?

**Comment by prj-** [comment]
No particular reason. They are superfluous, but should that impact the formatting?

**Comment by owenca** [comment]
> No particular reason. They are superfluous, but should that impact the formatting?

Yes, because it could be a chained cast if `numi` and `numj` are types and `HydHx` and `HydHx` are pointers.

**Comment by prj-** [comment]
I see. So no easy way to get this formatted the same in both cases?

**Comment by owenca** [comment]
No, because there's no way to tell clang-format that `numi` is _not_ a type.

**Comment by rymiel** [comment]
How common is a chained cast?

**Comment by owenca** [comment]
I would say not very common. @mydeveloperday added the support in https://reviews.llvm.org/D116592, which fixed #52979. Maybe we can narrow it down to chained _pointer_ types only.

**Comment by prj-** [comment]
I have removed most of our superfluous parentheses, so feel free to close this issue if you don't think this is appropriate.
